bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3775,6 +3779,7 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
   if ((lastExecutedSeqNum + 1) % checkpointWindowSize == 0) {
     checkpointNum = (lastExecutedSeqNum + 1) / checkpointWindowSize;
     stateTransfer->createCheckpointOfCurrentState(checkpointNum);
+    checkpoint_times_.start(checkpointNum);
```

Comment:
It appears that you are using checkpoint numbers here which are sequence numbers divided by window size. However, in the end method you are using sequence numbers directly. These will not match up.

Reply:
fixed